workflow-service/src/test/java/com/redhat/parodos/project/controller/ProjectControllerTest.java

```diff
@@ -0,0 +1,146 @@
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.util.List;
+import java.util.UUID;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ActiveProfiles("local")
+public class ProjectControllerTest extends ControllerMockClient {
+
+	@Autowired
+	private MockMvc mockMvc;
+
+	@MockBean
+	private ProjectServiceImpl projectService;
+
+	@Test
+	public void testCreateValidProject() throws Exception {
+		// given
+		ProjectRequestDTO project1DTO = new ProjectRequestDTO();
+		project1DTO.setName("project1");
+
+		ObjectMapper objectMapper = new ObjectMapper();
+		String json = objectMapper.writeValueAsString(project1DTO);
+
+		ProjectResponseDTO response = getSampleProject("project1");
+
+		Mockito.when(projectService.save(Mockito.any())).thenReturn(response);
+
+		// When
+		mockMvc.perform(this.postRequestWithValidCredentials("/api/v1/projects/").content(json)
+				.contentType(MediaType.APPLICATION_JSON)).andExpect(MockMvcResultMatchers.status().isCreated())
+				.andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+				.andExpect(MockMvcResultMatchers.jsonPath("$.id", Matchers.is(response.getId())))
+				.andExpect(MockMvcResultMatchers.jsonPath("$.name", Matchers.is(response.getName())));
+
+		// Then
+		Mockito.verify(projectService, Mockito.times(1)).save(Mockito.any());
+	}
+
+	@Test
+	public void testCreateValidProjectWithInvalidAuth() throws Exception {
+		// given
+		ProjectRequestDTO project1DTO = new ProjectRequestDTO();
+		project1DTO.setName("project1");
+		ObjectMapper objectMapper = new ObjectMapper();
+		String json = objectMapper.writeValueAsString(project1DTO);
+		// When
+		mockMvc.perform(this.postRequestWithInValidCredentials("/api/v1/projects/").content(json)
+				.contentType(MediaType.APPLICATION_JSON)).andExpect(MockMvcResultMatchers.status().isUnauthorized());
+
+		// Then
+		Mockito.verify(projectService, Mockito.times(0)).save(Mockito.any());
+	}
+
+	@Test
+	public void testListProjects() throws Exception {
+		ProjectResponseDTO project1DTO = getSampleProject("project1");
+		ProjectResponseDTO project2DTO = getSampleProject("project2");
+		Mockito.when(projectService.getProjects()).thenReturn(List.of(project1DTO, project2DTO));
+
+		// When
+		mockMvc.perform(this.getRequestWithValidCredentials("/api/v1/projects/"))
+				.andExpect(MockMvcResultMatchers.status().isOk())
+				.andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[0].id", Matchers.is(project1DTO.getId())))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[0].name", Matchers.is(project1DTO.getName())))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[1].id", Matchers.is(project2DTO.getId())))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[1].name", Matchers.is(project2DTO.getName())));
```

Comment:
I'd like to suggest testing the JSON array size as well.

Reply:
done